### ISSUE TITLE
Get product.info.delivery block issue.

### ISSUE THREAD

**Issue by choma828** [firegento/firegento-magesetup2#34]
I installed this extension on my magento, but it has an issue, when products list is in the cms homepage or other pages.

It seems you added 'product.info.delivery' block into only catalog_category_view.xml file, and no conditions for return null from getBlock('product.info.delivery' in https://github.com/firegento/firegento-magesetup2/blob/develop/Plugin/Catalog/ListProductPlugin.php#L37

I think it's better to update this like below.

```
    public function aroundGetProductDetailsHtml(
        \Magento\Catalog\Block\Product\ListProduct $subject,
        \Closure $proceed,
        \Magento\Catalog\Model\Product $product)
    {
        $deliveryBlock = $subject->getLayout()->getBlock('product.info.delivery');
        if($deliveryBlock) {
            $deliveryBlock->setProduct($product);
            $result = $proceed($product);
            if ((bool)$this->_scopeConfig->getValue(
                    'catalog/frontend/display_delivery_time',
                    \Magento\Store\Model\ScopeInterface::SCOPE_STORE
                )) {
    ;
                return $deliveryBlock->toHtml() . $result;
            }
        }
    }
```

**Comment by therouv** [comment]
Thanks for your suggestion - good idea! Unfortunately the code can not be used like this because in case  the block is null, the $proceed call won't be executed and hence no price output. 

I will optimize this code.

**Comment by wwenOne** [comment]
Hi, i have the same problem. Is there a usable solution?
 thx

**Comment by s1m0nde** [comment]
I guess I have the same problem. If so, it screwed up my create product page:
![createproductpage](https://cloud.githubusercontent.com/assets/6803289/17731480/c1d45ab4-646d-11e6-898c-6e30bb6b477a.jpg)
Another Magento2 installation with firegento (same version) had no such issues.

**Comment by therouv** [comment]
Fixed with merge of PR #39
